hkmc2/shared/src/test/mlscript/basics/ModuleMethods.mls

```diff
@@ -0,0 +1,67 @@
+class C with {
+  // regular method foo
+  fun foo(x) = x
+}
+
+module M with {
+  // module method foo
+  fun foo(x) = x
+}
+
+:e
+fun f1(module m)
+//│ ╔══[ERROR] Module parameters must have explicit types.
+//│ ║  l.12: 	fun f1(module m)
+//│ ╙──      	              ^
+
+:e
+fun f2[T](module m: T)
+//│ ╔══[ERROR] Module parameters must have concrete types.
+//│ ║  l.18: 	fun f2[T](module m: T)
+//│ ╙──      	                 ^^^^
+
+:e
+module N with {
+  fun f3(): M = M
+}
+//│ ╔══[ERROR] The return type of functions returning module values must be prefixed with module keyword.
+//│ ║  l.25: 	  fun f3(): M = M
+//│ ╙──      	      ^^^^^^^
+
+:e
+module N with {
+  fun f4[T](): module T = M
+}
+//│ ╔══[ERROR] Function returning module values must have concrete return types.
+//│ ║  l.33: 	  fun f4[T](): module T = M
+//│ ╙──      	      ^^^^^^^^^^^^^^^^^
+
+:e
+module N with {
+  fun f5(): M = M
+}
+//│ ╔══[ERROR] The return type of functions returning module values must be prefixed with module keyword.
+//│ ║  l.41: 	  fun f5(): M = M
+//│ ╙──      	      ^^^^^^^
+
+:e
+fun f6(module m: M)
+f6(new C)
+//│ ╔══[ERROR] Only module arguments (values) may be passed to module parameters.
+//│ ║  l.49: 	f6(new C)
+//│ ╙──      	       ^
+
+:e
+fun f7(): module M
+//│ ╔══[ERROR] Only module methods may return module values.
+//│ ║  l.55: 	fun f7(): module M
+//│ ╙──      	    ^^^^^^^^^^^^^^
+
+
+fun ok1(module m: M)
+
+module N with {
+  fun ok2(): module M = M
```

Comment:
Why not add test cases calling this function?

Reply:
The only rule in this PR regarding function call is "only module parameters may receive module arguments", and `ok2` does not involve any module parameters. So I think there's no need to call it.

Reply:
Sounds good, I'll add `ok1(N.ok2())`

Reply:
Added in another way: `ok1(M.self())`